### ISSUE TITLE
Please don't raise warning about snake case when struct has #[allow(non_snake_case)]

### ISSUE THREAD

**Issue by e-oz** [intellij-rust/intellij-rust#751]
Please don't raise warning about snake case when struct has #[allow(non_snake_case)] annotation.
example of code:

```
#[allow(non_snake_case)]
#[derive(Debug)]
pub struct HisciReportDetail {
  pub txtPharmacyGPOIDValue2: Option<String>,
  pub txtPharmacyGLNValue2: Option<String>,
  pub HealthIndustryNumber: Option<String>,
}
```

**Comment by alygin** [comment]
Unfortunately, ignoring the `allow` attributes is a common flaw of all the inspections now. I'll get back to this problem shortly.

**Comment by matklad** [comment]
I've added a suppression for the most common case of `allow(non_snake_case)` on a struct. Of course we wan't to handle all flavors of attributes eventually.
